addons/purchase_stock/tests/test_purchase_stock_report.py

```diff
@@ -364,7 +371,8 @@ def test_vendor_delay_report_without_backorder(self):
             [('partner_id', '=', self.partner.id)],
             ['product_id'],
             ['on_time_rate:sum', 'qty_on_time:sum', 'qty_total:sum'],
-        )['groups'][0]
-        self.assertEqual(data['qty_on_time:sum'], 6)
-        self.assertEqual(data['qty_total:sum'], 10)
-        self.assertEqual(data['on_time_rate:sum'], 60)
+        )['groups']
+        self.assertTrue(data, "Vendor delay report should return data even if product has no category")
```

Comment:
It would mean that, if `data` is falsy, it's for sure because of the missing category
We can't predict that
You can actually remove the new assert and let the existing ones as they are: thanks to the product you are using, the test will crash without the fix, so it's already ok